contracts/v2/arbitrum-orbit/FiatTokenArbitrumOrbitV2_2.sol

```diff
@@ -0,0 +1,128 @@
+     * Slot is hard-coded so that base USDC contracts can be further expanded without the possibility
+     * of slot collision.
+     */
+    bytes32
+        private constant L1_ADDRESS_SLOT = 0x54352c0d7cc5793352a36344bfdcdcf68ba6258544ce1aed71f60a74d882c191;
+
+    modifier onlyGateway() {
+        require(msg.sender == l2Gateway(), "ONLY_GATEWAY");
+        _;
+    }
+
+    /**
+     * @notice initialize the token
+     * @param l2Gateway_ child chain gateway this token communicates with
+     * @param l1Counterpart_ parent chain address of the USDC
+     */
+    function initializeArbitrumOrbit(address l2Gateway_, address l1Counterpart_)
+        external
+    {
+        // initializeV2_2 of FiatTokenV2_2 sets version to 3
+        require(_initializedVersion == 3);
+        require(l2Gateway_ != address(0), "INVALID_GATEWAY");
+        require(l2Gateway() == address(0), "ALREADY_INIT");
+        assembly {
+            sstore(L2_GATEWAY_SLOT, l2Gateway_)
+            sstore(L1_ADDRESS_SLOT, l1Counterpart_)
+        }
+    }
+
+    /**
+     * @notice Get the address of the child chain gateway connected to this token
+     */
+    function l2Gateway() public view returns (address _l2Gateway) {
+        assembly {
+            _l2Gateway := sload(L2_GATEWAY_SLOT)
+        }
+    }
+
+    /**
+     * @notice Get the address of the parent chain USDC
+     */
+    function l1Address() public override view returns (address _l1Address) {
+        assembly {
+            _l1Address := sload(L1_ADDRESS_SLOT)
+        }
+    }
+
+    /**
+     * @notice Mint tokens on child chain. Callable path is L1Gateway depositToken (which handles L1 escrow), which triggers L2Gateway, which calls this
+     * @param account recipient of tokens
+     * @param amount amount of tokens minted
+     */
+    function bridgeMint(address account, uint256 amount)
+        external
+        virtual
+        override
+        onlyGateway
+    {
+        mint(account, amount);
+    }
+
+    /**
+     * @notice Burn tokens on child chain.
+     * @dev only the token bridge gateway can call this
+     * @param account owner of tokens
+     * @param amount amount of tokens burnt
+     */
+    function bridgeBurn(address account, uint256 amount)
+        external
+        virtual
+        override
+        onlyGateway
+    {
+        _burn(account, amount);
+    }
+
+    /**
+     * @notice Allows a minter to burn some of its own tokens.
+     * @dev Function 'burn' in FiatTokenV1 is unaccessible from here in the same call due to being external,
+     *      and also it operates on msg.sender as burn account. Burn logic is copied over here into private function and burn account is provided.
+     * @dev The caller must be a minter, must not be blacklisted, and the amount to burn
+     * should be less than or equal to the account's balance.
+     * @param _amount the amount of tokens to be burned.
+     */
+    function _burn(address _account, uint256 _amount)
```

Comment:
we can use similar approach I described for `mint`, but since `FiatTokenV1.burn` does not support burn from this will need to use `_transfer` to pull token from the user and then `burn`

Reply:
Due to token address being blacklisted in one of the initializers, burn/mint logic is still duplicated in internal functions